block/daIncluder.go

```diff
@@ -0,0 +1,67 @@
+package block
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+)
+
+// DAIncluderLoop is responsible for advancing the DAIncludedHeight by checking if blocks after the current height
+// have both their header and data marked as DA-included in the caches. If so, it calls setDAIncludedHeight.
+func (m *Manager) DAIncluderLoop(ctx context.Context) {
```

Comment:
do we also want this to be used in conjunction with pending blocks, it could help be a way to see how far ahead we are

Reply:
not sure how pending blocks are related here, can you please expand?

Reply:
high level thought in the max_pending_blocks design is that we dont know realistically how far ahead we are, only can make assumptions on how much is pending from my understanding. With daincluded we could know exactly how far ahead we really are

Reply:
max_pending_blocks exists so the sequencer node stops producing rollup blocks if it's not able to submit pending headers to the DA layer, likely in the case, when it loses the connection to the DA layer. Note that the DAIncluded height refers to a rollup block height which we can mark final when both the header and block height are seen on the DA layer. Not sure how it refers to how far ahead we are still 🤔

Reply:
Documented in an issue: https://github.com/rollkit/rollkit/issues/2252